templates-global/hero.php

```diff
@@ -22,17 +23,43 @@
 	$hero_highlight = '';
 }
 
+/**
+ * The text color of the hero title is usually black. However, when 'none' is
+ * chosen, we allow the user to choose between black or white text to best fit
+ * with the image.
+ */
+if ( 'white' == $title_color && 'none' == $hero_highlight || empty( $hero_highlight ) ) {
+	$title_color_class = 'text-white';
+} else {
+	$title_color_class = '';
+}
+
+switch ( $hero_size ) {
+	case 'small':
+		$hero_size_class = 'uds-hero-sm';
+		break;
+	case 'medium':
+		$hero_size_class = 'uds-hero-md';
+		break;
+	case 'large':
+		$hero_size_class = 'uds-hero-lg';
+		break;
+	default:
+		// Should there be a problem, default to medium size.
+		$hero_size_class = 'uds-hero-md';
+		break;
+}
 
 // Check for a hero bg image and if present build the hero. Otherwise show the title of the page.
 if ( ! empty( $hero_asset_url ) ) :
 	?>
-<div class="uds-hero <?php echo $hero_type; ?>" style="background-image: linear-gradient(180deg, #19191900 0%, #191919c9 100%), url('<?php echo wp_kses( $hero_asset_url, wp_kses_allowed_html( 'strip' ) ); ?>')">
-	<div class="container uds-hero-container">
+<div class="uds-hero <?php echo $hero_size_class; ?>" style="background-image: linear-gradient(180deg, #19191900 0%, #191919c9 100%), url('<?php echo wp_kses( $hero_asset_url, wp_kses_allowed_html( 'strip' ) ); ?>')">
+	<div class="container uds-hero-container lazyloaded">
 		<?php
 		if ( ! empty( $hero_title ) ) :
 			?>
 		<h1 class="col-md-8">
-			<span class="<?php echo $hero_highlight; ?>"><?php echo wp_kses( $hero_title, wp_kses_allowed_html( 'strip' ) ); ?></span>
+			<span class="<?php echo $hero_highlight; ?> <?php echo $title_color_class; ?>"><?php echo wp_kses( $hero_title, wp_kses_allowed_html( 'strip' ) ); ?></span>
 		</h1>
```

Comment:
Since it's either **$hero_highlight** or **$title_color_class**.
We can create one var called **$hero_title_class**, by default it takes the value of **$hero_highlight**, and would take the value of the **$title_color_class** if it went through the if condition `if ( 'white' == $title_color) or black`.